Per os config feature

marcmengel opened the pull request:
This is an implementation for Feature request #31026/#2700, and a redo of #16834 and #13686, fixed for latest version.

marcmengel committed:
per-os config

marcmengel committed:
per-os config

marcmengel committed:
fix for newer version

marcmengel committed:
fix for newer version

marcmengel committed:
make flake8 happy

marcmengel committed:
make flake8 happy

marcmengel committed:
make black happy

marcmengel commented:
rebased, black formatted.

scheibelp commented on lib/spack/spack/config.py:
```diff
@@ -773,6 +782,7 @@ def _add_command_line_scopes(cfg, command_line_scopes):
         name = "cmd_scope_%d" % i
         cfg.push_scope(ImmutableConfigScope(name, path))
         _add_platform_scope(cfg, ImmutableConfigScope, name, path)
+        _add_os_scope(cfg, ImmutableConfigScope, name, path)
```
I think you'll also want to do this in the other location where `_add_platform_scope` is called, in `_config` (otherwise only config directories added with `spack -C ...` will include OS scopes, and not e.g. `~/.spack/`).

marcmengel replied:
I thought I'd done that... Thanks.   Pushed.

marcmengel committed:
second scope add location

marcmengel committed:
use os.path.join so windows test doesn't barf

tgamblin committed:
Add `concretise` and `containerise` aliases for our UK users

From a user:

> Aargh.
> ```
> ==> Error: concretise is not a recognized Spack command or extension command; check with `spack commands`.
> ```

To make things easier for our friends in the UK, this adds `concretise` and
`containerise` aliases for the `spack concretize` and `spack containerize` commands.

- [x] add aliases
- [x] update completions

tgamblin committed:
completion: add alias handling

Bash completion is now smarter about handling aliases. In particular, if all completions
for some input command are aliased to the same thing, we'll just complete with that thing.

If you've already *typed* the full alias for a command, we'll complete the alias.

So, for example, here there's more than one real command involved, so all aliases are
shown:

```console
$ spack con
concretise    concretize    config        containerise  containerize
```

Here, there are two possibilities: `concretise` and `concretize`, but both map to
`concretize` so we just complete that:

```console
$ spack conc
concretize
```

And here, the user has already typed `concretis`, so we just go with it as there is only
one option:

```console
 spack concretis
concretise
```

svengoldberg committed:
Include t8code package (#39391)

* t8code: Create new spack package

Co-authored-by: Bernhard Kaindl <bernhardkaindl7@gmail.com>

becker33 committed:
Add amg2023 package; remove amg package (#39105)

Add amg2023 package

Consolidate existing amg and amg2013 packages (they reference the
same code) under the amg2013 name to minimize confusion between
amg2023 and amg2013.

Co-authored-by: Riyaz Haque <haque1@llnl.gov>

rbberger committed:
slurm: add version 23-02-3-1 and 23-02-4-1 (#39598)

rbberger committed:
lammps: simplify url_for_version logic (#39554)

Use the stable_versions variable to check stable versions

brenthuisman committed:
Arbor: version added: v0.9.0 (#39374)

jcortial-safran committed:
Make lcov dependency on perl explicit (#39485)

aumuell committed:
ffmpeg: apply upstream fix for build with binutils 2.41 (#39392)

While spack does not yet provide binutils 2.41, they might still be
installed. However, building ffmpeg on x86_64 fails with multiple errors like
this:
./libavcodec/x86/mathops.h:125: Error: operand type mismatch for `shr'

also reported here: https://trac.ffmpeg.org/ticket/10405

jcortial-safran committed:
Add package mpi-test-suite (#39487)

alanorth committed:
perl-http-message: add dependency on perl-clone (#39567)

Clone is a hard dependency as of HTTP-Message v6.44. This causes
problems in packages like Roary which depend on perl-http-message:

    $ spack load roary
    $ roary
    Use of uninitialized value in require at /var/scratch/spack/opt/spack/linux-centos8-x86_64_v3/gcc-8.5.0/perl-http-message-6.44-lzp5th4jddd3gojkjfli4hljgem2nl26/lib/perl5/HTTP/Headers.pm line 8.
    Can't locate Clone.pm in @INC (you may need to install the Clone module) (@INC contains:  /home/aorth/lib ...

See: https://github.com/libwww-perl/HTTP-Message/blob/master/Changes

rem1776 committed:
fms: add v2023.02 and v2023.01.01 (#39434)

* Add 2023.02 and 2023.01.01 with variant for mpp_io and fms_io deprecation

Co-authored-by: rem1776 <Ryan.Mulhall@noaa.gov>

dependabot[bot] committed:
build(deps): bump sphinx from 7.2.2 to 7.2.3 in /lib/spack/docs (#39603)

Bumps [sphinx](https://github.com/sphinx-doc/sphinx) from 7.2.2 to 7.2.3.
- [Release notes](https://github.com/sphinx-doc/sphinx/releases)
- [Changelog](https://github.com/sphinx-doc/sphinx/blob/master/CHANGES)
- [Commits](https://github.com/sphinx-doc/sphinx/compare/v7.2.2...v7.2.3)

---
updated-dependencies:
- dependency-name: sphinx
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

alalazo committed:
Remove leftover attributes from parser (#39574)

#35042 introduced lazy hash parsing, but didn't remove a
few attributes from the parser that were needed only for
concrete specs

This commit removes them, since they are effectively
dead code.

haampie committed:
rebuild-index: fix race condition by avoiding parallelism (#39594)

eugeneswalker committed:
h5bench: add e4s tag (#39595)

haampie committed:
zlib-ng: build system patches (#39559)

s-sajid-ali committed:
add magma variant to strumpack (#39224)

* add magma variant to strumpack
* clarify conflicts to be excuive to rocm/cuda, without it +magma+cuda fails as it is ~rocm
   modified:   var/spack/repos/builtin/packages/strumpack/package.py
* add missing depends_on for magma variant

adamjstewart committed:
py-sphinx: add v7.2 (#39513)

fsimonis committed:
preCICE: prepare for version 3 (#39486)

wdconinc committed:
py-particle: new versions through 0.23.0 (#39547)

* py-particle: new versions through 0.23.0

* py-particle: depends_on py-deprecated

MattToast committed:
[py-smartsim] New Package (#39306)

* Create a smartsim package

* rm ss 0.4.2

* no py upper bound, add build dep

* add setup_build_env

* add comment to find ml deps lower bounds

* Apply suggestions from code review

Correct dep versions, use `python_purelib`

Co-authored-by: Adam J. Stewart <ajstewart426@gmail.com>

* remove the cuda/rocm vars

* point editors to bin deps version constraints

* Apply suggestions from code review

Loosen `py-smartredis` constraint, enforce `setup.cfg` py version

Co-authored-by: Adam J. Stewart <ajstewart426@gmail.com>

* style

* rm rai lower bound

* lower bound setuptools

Co-authored-by: Adam J. Stewart <ajstewart426@gmail.com>

---------

Co-authored-by: Adam J. Stewart <ajstewart426@gmail.com>

amd-toolchain-support committed:
AOCL spack recipes for 4.1 release w. new AOCL-Utils library (#39489)

* AOCC and AOCL spack recipes for 4.1 release
* Fix broken checksum
* remove blank line
* Add missing `@when` for 4.1 only function

---------

Co-authored-by: vijay kallesh <Vijay-teekinavar.Kallesh@amd.com>

jthies committed:
phist: new version 1.12.0 (#39616)

rbberger committed:
legion: depend on cray-pmi when using gasnet and cray-mpich (#39614)

adamjstewart committed:
py-sphinx: add v7.2.3 (#39612)

dependabot[bot] committed:
build(deps): bump actions/checkout from 3.5.3 to 3.6.0 (#39617)

Bumps [actions/checkout](https://github.com/actions/checkout) from 3.5.3 to 3.6.0.
- [Release notes](https://github.com/actions/checkout/releases)
- [Changelog](https://github.com/actions/checkout/blob/main/CHANGELOG.md)
- [Commits](https://github.com/actions/checkout/compare/c85c95e3d7251135ab7dc9ce3241c5835cc595a9...f43a0e5ff2bd294095638e18286ca9a3d1956744)

---
updated-dependencies:
- dependency-name: actions/checkout
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

rbberger committed:
legion: drop cray-pmi dependency (#39620)

There are other ways to enforce cray-pmi being loaded in environments
that use cray-mpich. This avoids breaking environments where this was
already the case and avoids forcing them to declare an external.

heerener committed:
arrow: explicitly disable RE2 and UTF8PROC flags when not in dependencies (#35353)

Co-authored-by: Harmen Stoppels <harmenstoppels@gmail.com>

adamjstewart committed:
py-scipy: add v1.11.2 (#39515)

lorisercole committed:
intel-oneapi-mkl: add hpcx-mpi to the list of supported MPI libs (#39625)

Co-authored-by: Loris Ercole <a-lercole@microsoft.com>

Thomas-Ulrich committed:
add memalign option to petsc to be used with tandem (#37282)

jamessmillie committed:
Windows symlinking support (#38599)

This reapplies 66f7540, which adds supports for hardlinks/junctions on
Windows systems where developer mode is not enabled.

The commit was reverted on account of multiple issues:

* Checks added to prevent dangling symlinks were interfering with
  existing CI builds on Linux (i.e. builds that otherwise succeed were
  failing for creating dangling symlinks).
* The logic also updated symlinking to perform redirection of relative
  paths, which lead to malformed symlinks.

This commit fixes these issues.

bvanessen committed:
LBANN package: strip hostname on non-LC systems. (#39601)

prckent committed:
QMCPACK: add v3.17.1 (#39626)

msimberg committed:
Add gperftools 2.12 (#39623)

naromero77 committed:
llvm: fix `+lldb` optional deps and add `lua` variant (#39579)

* llvm: fix +lldb optional deps, add lua variant

* use good old conflicts

---------

Co-authored-by: Harmen Stoppels <me@harmenstoppels.nl>

tgymnich committed:
Update package.py (#39619)

LydDeb committed:
py-dask: use py-jinja2 versions from 2.10.3 (#39631)

In depends_on py-jinja2, use versions from 2.10.3, instead of strict equality, according to the pyproject.toml of dask 2023.4.1

DarylGrunau committed:
py-scipy: re-apply scipy/scipy#14935 workaround (#39555)

* re-apply scipy/scipy#14935 workaround

* restrict to %intel and eliminate Version() check

---------

Co-authored-by: Daryl W. Grunau <dwg@lanl.gov>

johnwparent committed:
CMake: add versions 3.26.4|5 and 3.27.* (#39628)

Co-authored-by: Harmen Stoppels <me@harmenstoppels.nl>

scheibelp committed:
External ROCm: add example configuration (#39602)

* add an example of an external rocm configuration
* include more info
* generalize section to all GPU support

tgamblin committed:
completion: make bash completion work in zsh

`compgen -W` does not behave the same way in zsh as it does in bash; it seems not to
actually generate the completions we want.

- [x] add a zsh equivalent and `_compgen_w` to abstract it away
- [x] use `_compgen_w` instead of `compgen -W`

naromero77 committed:
llvm: ensure runtimes set rpaths (#39641)

Ensure that CMAKE_INSTALL_RPATH_USE_LINK_PATH is propagated to sub-make when building runtimes.

ChristopherChristofi committed:
py-igraph: new package with version 0.10.6 (#39179)

haampie committed:
libuv: add versions up to v1.46.0  (#39647)

haampie committed:
libarchive: new versions (#39646)

stomov committed:
magma: add v2.7.2 (#39645)

haampie committed:
libarchive: use openssl by default (#39640)

haampie committed:
Remove mxnet from darwin aarch64 since theres no supported release (#39644)

qwertos committed:
py-scikit-optimize: add v0.9.0 and move gptune patch (#38738)

Co-authored-by: liuyangzhuan <liuyangzhuan@gmail.com>

haampie committed:
Fix a few issues with git versions / moving targets in packages (#39635)

alalazo committed:
GnuPG: add v2.4.3 (#39654)

Also updates a few dependencies

iamashwin99 committed:
octopus: specify the scalapack library not the directory (#39109)

* octopus: specify the scalapack library not the directory
* octopus: use spec.scalapack.libs.ld_flags

tgamblin committed:
security: change `SECURITY.md` to recommend GitHub's private reporting (#39651)

GitHub's beta private security issue reporting feature is enabled on the Spack repo now,
so we can change `SECURITY.md` to recommend using it instead of `maintainers@spack.io`.

- [x] Update `SECURITY.md` to direct people to the GitHub security tab.
- [x] Update working in `SECURITY.md` to say "last two major releases" with a link to
      the releases page, instead of explicitly listing release names. This way we don't have
      to update it (which we keep forgetting to do).

mathomp4 committed:
mapl: add 2.40, deprecate old versions (#39615)

* Add MAPL 2.40, deprecated old versions
* Undo some variant changes
* style fixes
* Make variants reflect Cmake options
* Remove pnetcdf variant
* Clean up depends_on

lorisercole committed:
kokkos: avoid setting multiple `cuda_arch` (#38723)

The kokkos package does not support building with cuda with multiple
`cuda_arch`. An error is raised if multiple `cuda_arch` are specified.

vanderwb committed:
singularityce: arg handling fix (#39663)

kwryankrattiger committed:
Patch VTK to enable python 3.8 in VTK 8.2 (#38735)

* VTK: Add patch for python 3.8 support

* CI: Re-enable VisIt in CI

* Configure spec matrix for stack with VisIt

* Add pugixml dep for 8.2.0

* Make VTK and ParaView consistent on proj dep

* OpenMPI 3: provides MP support by default

* Add details on proj dep in ParaView

* Add python 3.8 to test mock repo

* Patches to get VisIt VTK interface

* CI: Disable VisIt with GUI in DAV

blue42u committed:
modules: copy matched config to prevent bleed (#39421)

adamjstewart committed:
Add style tool to fix fish file formatting (#39155)

haampie committed:
fmt: 10.1.1 (#39664)

dependabot[bot] committed:
build(deps): bump docker/setup-buildx-action from 2.9.1 to 2.10.0 (#39669)

Bumps [docker/setup-buildx-action](https://github.com/docker/setup-buildx-action) from 2.9.1 to 2.10.0.
- [Release notes](https://github.com/docker/setup-buildx-action/releases)
- [Commits](https://github.com/docker/setup-buildx-action/compare/4c0219f9ac95b02789c1075625400b2acbff50b1...885d1462b80bc1c1c7f0b00334ad271f09369c55)

---
updated-dependencies:
- dependency-name: docker/setup-buildx-action
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps): bump sphinx from 7.2.3 to 7.2.4 in /lib/spack/docs (#39668)

Bumps [sphinx](https://github.com/sphinx-doc/sphinx) from 7.2.3 to 7.2.4.
- [Release notes](https://github.com/sphinx-doc/sphinx/releases)
- [Changelog](https://github.com/sphinx-doc/sphinx/blob/master/CHANGES)
- [Commits](https://github.com/sphinx-doc/sphinx/compare/v7.2.3...v7.2.4)

---
updated-dependencies:
- dependency-name: sphinx
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

haampie committed:
Automated git version fixes (#39637)

Use full length commit sha instead of short prefixes, to improve
reproducibility (future clashes) and guard against compromised repos and
man in the middle attacks.

Abbreviated commit shas are expanded to full length, to guard against future
clashes on short hash. It also guards against compromised repos and
man in the middle attacks, where attackers can easily fabricate a malicious
commit with a shasum prefix collision.

Versions with just tags now also get a commit sha, which can later be used to
check for retagged commits.

haampie committed:
Revert "Add style tool to fix fish file formatting (#39155)" (#39680)

This reverts commit 70c71e8f935b3c5152036a993295df13c5fe69ca.

LydDeb committed:
py-distributed: use py-jinja2 versions from 2.10.3 (#39658)

In depends_on py-jinja2, use versions from 2.10.3, instead of strict equality, according to the pyproject.toml of py-distributed 2023.4.1

adamjstewart committed:
py-lightly: add v1.4.17 (#39686)

qwertos committed:
[py-gradio] New package (#39304)

* [py-gradio] New package

* [py-gradio] added deps from requirements.txt

* [py-gradio] py-pillow -> pil

mathomp4 committed:
mapl: Fix pflogger dependency (#39683)

adamjstewart committed:
py-torchmetrics: add v1.1.1 (#39685)

qwertos committed:
[py-tiktoken] added verison 0.3.1 (#39677)

asartori86 committed:
justbuild: add v1.2.1 (#39679)

jedwards4b committed:
Jedwards/parallelio262 (#39676)

* add parallelio@2.6.2
* add comments

qwertos committed:
[mathematica] added version 13.0.1 (#39675)

G-Ragghianti committed:
Added release 2023.08.25 (#39665)

payerle committed:
boxlib: Add BL_USE_PARTICLES cmake var (#39660)

Set BL_USE_PARTICLES to 1, which should case boxlib build to include Particles classes
according to CMakeLists.txt.

This seems to fix #18172
The aforementioned error seems to occur in cmake phase while processing
CMakeLists.txt in Src/C_ParticleLib, and appears to be due to the
variable containing the list of src files for the add_library() call
being empty unless BL_USE_PARTICLES is set to 1.

adamjstewart committed:
libpng: add libs property (#39638)

lpottier committed:
Upgrading py-pika versions (#39670)

* Upgraded py-pika versions

Signed-off-by: Loïc Pottier <pottier1@llnl.gov>

* Upgraded py-pika versions

Signed-off-by: Loïc Pottier <pottier1@llnl.gov>

* added variants and adding setuptools constraint

Signed-off-by: Loïc Pottier <pottier1@llnl.gov>

* missing setuptools for older versions

Signed-off-by: Loïc Pottier <pottier1@llnl.gov>

---------

Signed-off-by: Loïc Pottier <pottier1@llnl.gov>

dyokelson committed:
amg2023 package: cmake build (#39618)

Switch from makefile to CMake-based build. CMake support is currently
in a specific branch of the amg2023 repo, so add this branch as a
version in the package.

myan-rarcellino-nnl committed:
py-parsl: add v2023.08.21 (#39642)

* py-parsl: add v2023.08.21

* fix style

* add missing deps

greenc-FNAL committed:
[procps] New versions (#38650)

* [procps] New versions 3.3.14, 3.3.3.15..3.3.17, 4.0.0..4.0.3

* Put upper bound on patch application

* [Patch was
  merged](https://gitlab.com/procps-ng/procps/-/merge_requests/189) for
  upcoming 4.0.4.

adamjstewart committed:
py-llvmlite: add Python version requirements (#39711)

johnwparent committed:
Windows: oneapi/msvc consistency (#39180)

Currently, OneAPI's setvars scripts effectively disregard any arguments
we're passing to the MSVC vcvars env setup script, and additionally,
completely ignore the requested version of OneAPI, defaulting to whatever
the latest installed on the system is.

This leads to a scenario where we have improperly constructed Windows
native development environments, with potentially multiple versions of
MSVC and OneAPI being loaded or called in the same env. Obviously this is
far from ideal and leads to some fairly inscrutable errors such as
overlapping header files between MSVC and OneAPI and a different version
of OneAPI being called than the env was setup for.

This PR solves this issue by creating a structured invocation of each
relevant script in an order that ensures the correct values are set in
the resultant build env.

The order needs to be:

1. MSVC vcvarsall
2. The compiler specific env.bat script for the relevant version of
   the oneapi compiler we're looking for. The root setvars scripts seems
   to respect this as well, although it is less explicit
3. The root oneapi setvars script, which sets up everything else the
   oneapi env needs and seems to respect previous env invocations.

pmccormick committed:
llvm: fix for Flang variant due to exception handling (#36171)

* The flang project does not support exceptions enabled in the core llvm
library (and developer guidelines explicitly state they should not be
used).  For this reason, when the flang variant is selected,
LLVM_ENABLE_EH needs to be disabled.  In the current main branch of
llvm (and thus future releases), enabling flang and setting
LLVM_ENABLE_EH will cause the overall build to fail.

* Update var/spack/repos/builtin/packages/llvm/package.py

Co-authored-by: Harmen Stoppels <harmenstoppels@gmail.com>

* fix syntax

---------

Co-authored-by: Harmen Stoppels <harmenstoppels@gmail.com>
Co-authored-by: Satish Balay <balay@mcs.anl.gov>

haampie committed:
ASP-based solver: add hidden mode to ignore versions that are moving targets, use that in CI (#39611)

Setting the undocumented variable SPACK_CONCRETIZER_REQUIRE_CHECKSUM
now causes the solver to avoid accounting for versions that are not checksummed.

This feature is used in CI to avoid spurious concretization against e.g. develop branches.

rasolca committed:
dla-future: add v0.2.0 (#39705)

haampie committed:
Remove a few local imports in tests (#39719)

dependabot[bot] committed:
build(deps): bump sphinx from 7.2.4 to 7.2.5 in /lib/spack/docs (#39716)

Bumps [sphinx](https://github.com/sphinx-doc/sphinx) from 7.2.4 to 7.2.5.
- [Release notes](https://github.com/sphinx-doc/sphinx/releases)
- [Changelog](https://github.com/sphinx-doc/sphinx/blob/master/CHANGES)
- [Commits](https://github.com/sphinx-doc/sphinx/compare/v7.2.4...v7.2.5)

---
updated-dependencies:
- dependency-name: sphinx
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

alalazo committed:
environment: improve spack remove matching (#39390)

search for equivalent specs, not for equal strings when selecting a spec to remove.

toxa81 committed:
[NVHPC] add a possibility to control default CUDA version (#38909)

* add a possibility to control default cuda version

* fix stype

* style fix

* resolve comment

* resolve comment

* Fix style in nvhpc package.py

---------

Co-authored-by: antonk <antonk@cscs.ch>
Co-authored-by: Mikael Simberg <mikael.simberg@iki.fi>

adamjstewart committed:
py-pandas: add v2.1.0 (#39707)

johnwparent committed:
"spack config add": allow values with a ":" (#39279)

If you wanted to set a configuration option like
`config:install_tree:root` to "C:/path/to/config.yaml", Spack  had
trouble parsing this because of the ":" in the value. This adds
logic to allow using quotes to enclose the value, so you can add
`config:install_tree:root:"C:/path/to/config.yaml"`.

Configuration keys should never contain a quote character, so the
presence of any quote is taken to mean that the rest of the string
is specifying the value.

johnwparent committed:
Windows: fix pwsh env activate/deactivate; load/unload (#39118)

These commands are currently broken on powershell (Windows) due to
improper use of the InvokeCommand commandlet and a lack of direct
support for the `--pwsh` argument in `spack load`, `spack unload`,
and `spack env deactivate`.

qwertos committed:
[py-preshed] added version 3.0.8 (#38969)

* [py-preshed] added version 3.0.8

* [py-preshed] reordered for consistancy

qwertos committed:
py-dataclasses-json: add new package (#39493)

* [py-dataclasses-json] New package

* [py-dataclasses-json] limiting py-poetry-core versions

haampie committed:
glibc: add package (#39695)

haampie committed:
Set backtrace in ci (#39737)

dependabot[bot] committed:
build(deps): bump pytest from 7.4.0 to 7.4.1 in /lib/spack/docs (#39790)

Bumps [pytest](https://github.com/pytest-dev/pytest) from 7.4.0 to 7.4.1.
- [Release notes](https://github.com/pytest-dev/pytest/releases)
- [Changelog](https://github.com/pytest-dev/pytest/blob/main/CHANGELOG.rst)
- [Commits](https://github.com/pytest-dev/pytest/compare/7.4.0...7.4.1)

---
updated-dependencies:
- dependency-name: pytest
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps): bump actions/checkout from 3.6.0 to 4.0.0 (#39789)

Bumps [actions/checkout](https://github.com/actions/checkout) from 3.6.0 to 4.0.0.
- [Release notes](https://github.com/actions/checkout/releases)
- [Changelog](https://github.com/actions/checkout/blob/main/CHANGELOG.md)
- [Commits](https://github.com/actions/checkout/compare/f43a0e5ff2bd294095638e18286ca9a3d1956744...3df4ab11eba7bda6032a0b82a6bb43b11571feac)

---
updated-dependencies:
- dependency-name: actions/checkout
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

wdconinc committed:
clhep: support variant cxxstd=20 (#39768)

pabloaledo committed:
Add maintainers to some bioconductor packages (#39801)

Signed-off-by: Pablo <pablo.aledo@seqera.io>

msimberg committed:
hpx-kokkos: add 0.4.0 and other minor changes (#39723)

* Add hpx-kokkos 0.4.0

* Make git global package property in hpx-kokkos instead of having it version-specific

* Add variant for choosing future type in hpx-kokkos

* Add support for testing hpx-kokkos

alalazo committed:
Add type-hints to the `spack.detection` package (#39803)

wdconinc committed:
boost: support variant cxxstd=20 (#39769)

jmcarcell committed:
podio, edm4hep and lcio: add lib and lib64 to LD_LIBRARY_PATH (#37881)


Co-authored-by: jmcarcell <jmcarcell@users.noreply.github.com>

wdconinc committed:
compilers/gcc.py: support cxx{20,23}_flag (#39777)

wdconinc committed:
xerces-c: support variant cxxstd=20 (#39784)

haampie committed:
Revert ""spack config add": allow values with a ":" (#39279)" (#39825)

This reverts commit b72a268bc565d370bd64589e86929acfd73c08a1.

msimberg committed:
umpire: Add 2022.10.0 (#37998)

* Add umpire 2022.10.0
* Add patches to umpire@2022.10.0
* Add axom 0.8.0 and 0.8.1
* Remove Umpire 2023.06.0

adamjstewart committed:
CMakePackage: remove -DBUILD_TESTING flag (#37967)

lmagdanello committed:
new package: Yade (#39786)

* feat: yade package
* fix: dependencies for Yade
* feat: yade versions
* fix: cmake_args and version_url
* fix: Spack license identifier
* fix: black formatter
* fix: black would be reformatted? yes!
* feat: add maintainer

---------

Co-authored-by: leo magdanello <araujo.leonardo@luizalabs.com>

streeve committed:
cabana: fix gtest dependency (#39811)

matheushms committed:
Add cmdlime package (#39835)

* Add cmdlime package
* remove the blank line at the end of file

haampie committed:
gitlab: set cache and config dir in top-level file (#39829)

edwardhartnett committed:
g2: add dependencies, build options, and v3.4.6 release (#39717)

* adding bacio to g2 dependencies
* edited documentation
* added version 3.4.6
* starting with 3.4.6, can use any version of jasper
* adding w3emc dependency for versions up to 3.4.5
* removed t-brown as maintainer at his request

streeve committed:
exaca: add version 1.2 (#39810)

aurianer committed:
Add pika 0.18.0 (#39842)

jack-morrison committed:
libfabric: Add versions 1.18.2, 1.19.0 (#39757)

johnwparent committed:
"spack config add": support values with ":" (#39831)

This is a fixed version of b72a268

* That commit would discard the final key component (so if you set
  "config:install_tree:root", it would discard "root" and just set
  install tree).
* When setting key:"value", with the quotes, that commit would
  discard the quotes, which would confuse the system if adding a
  value like "{example}" (the "{" character indicates a dictionary).
  This commit retains the quotes.

dominichofer committed:
Handles case of empty craype_dir (#39371)

The craype_dir folder can be empty. It happens to be the case on Arolla and Tsa at CSCS.

sxs-bot committed:
spectre: add v2023.07.29 (#39145)

Co-authored-by: sxs-bot <sxs-bot@users.noreply.github.com>

broderickpt committed:
FFTX: replace extends with variants (#39701)

The spiral-software package had a number of extensions, but does not
work unless they actually exist in the spiral software prefix (creating
a view is not sufficient). With the removal of "spack activate"
(different from "spack env activate"), a new approach is needed to
support optional components of `spiral-software` . This commit updates
the spiral-software package to copy the dependency installations into
its own prefix.

This commit also adds versions for `fftx` and `spiral-software`, as
well as an optional `spiral-software-jit` package.

haampie committed:
musl: bump (#39838)

kjrstory committed:
openfoam-org: add variant scotch (#39841)

simonpintarelli committed:
sirius: simplify hip architectures, fix umpire dependency (#39776)

jcbowden committed:
new damaris package updated to use Daamris v.1.9.0 and including py-mpi4py dependency when +python specified (#39774)

jgaeb committed:
r-humaniformat: Add R `humaniformat` package (#39761)

* r-humaniformat: Add r-humaniformat.
* r-humaniformat: Remove blank line at end of file.
* Fix broken link in description.

jgaeb committed:
r-sets: Add R `sets` package (#39760)

* r-sets: Add r-sets.
* r-sets: Fix R dependency for first version.
* r-sets: Remove package versions for which a compatible version of R cannot be installed.

jgaeb committed:
Add libpostal Package (#39758)

* Add libpostal.
* Add back Spack license.

RemiLacroix-IDRIS committed:
Eigen: Fix patch URL (#39746)

vanderwb committed:
Use correct hash for 3.0 (#39743)

kuberry committed:
xyce: add version 7.7.0 and specialize patch (#39696)

payerle committed:
cgns: Add -fPIC to Fortan/C compilation flags (#39700)

* cgns: Add -fPIC to Fortan/C compilation flags
  This should fix #39699
* cgns: Add -fPIC compilation flags when +fortran
  Incorporating suggestions of @aumuell with addition of making +pic required when +fortran

payerle committed:
seacas: Set TPL_ENABLE_Pthread=ON when +thread_safe (#39703)

* seacas: Set TPL_ENABLE_Pthread=ON when +thread_safe
  This should fix #39702
  Basically, following suggestion of error message and setting
  TPL_ENABLE_Pthread to the value of the boolean spack variant thread_safe
* seacas: Fix style issue
  Add space after comment #

RemiLacroix-IDRIS committed:
CUDA 10.1/10.2: Fix conflicts with PGI 19+. (#39747)

The condition probably did not get updated when the behavior of the version specifier changed.

jgaeb committed:
r-tinytiger: Add tinytiger R package. (#39828)

* r-tinytiger: Add tinytiger R package.
* Fix homepage.

---------

Co-authored-by: Tamara Dahlgren <35777542+tldahlgren@users.noreply.github.com>

glentner committed:
duckdb: add v0.8.1 (#39851)

matheushms committed:
Add Figcone Package (#39804)

* Add Figcone Package
* Fix the style
* add url
* fix style removing whitespace of a blanck line
* remove versions

jgaeb committed:
r-functional: Add the functional R package. (#39826)

Co-authored-by: Tamara Dahlgren <35777542+tldahlgren@users.noreply.github.com>

jgaeb committed:
r-furrr: Add furrr R package. (#39814)

Co-authored-by: Tamara Dahlgren <35777542+tldahlgren@users.noreply.github.com>

aumuell committed:
lua: unzip also required during build (#39802)

Unzip required during build otherwise configuration of luarocks fails during installation of lua@5.3.6 on ubuntu 22.04 w/o system unzip.

haampie committed:
move determine_number_of_jobs into spack.util.cpus, use it in concretize (#37620)

vsoch committed:
Automated deployment to update package flux-security 2023-09-07 (#39855)

Co-authored-by: github-actions <github-actions@users.noreply.github.com>

vsoch committed:
Automated deployment to update package flux-core 2023-09-07 (#39856)

Co-authored-by: github-actions <github-actions@users.noreply.github.com>

gardner48 committed:
add SUNDIALS v6.6.1 (#39853)

njansson committed:
neko: add v0.6.1 and fix package (#39751)

* neko: add v0.6.1 and fix package
* Change to a conditional variant for gslib

RMeli committed:
Allow to build latest released version of CP2K with CMake (#39782)

* allow to build latest version of cp2k with cmake
* apply black

streeve committed:
Make Cabana a Cuda/ROCm package (matching ArborX) (#39809)

eflumerf committed:
epics-base package: use the Spack compiler wrappers (#39752)

Edit configuration file to use the Spack compiler wrappers.

kwryankrattiger committed:
CI Timing Statistics (#38598)

* Write timing information for installs from cache

* CI: aggregate and upload install_times.json to artifacts

* CI: Don't change root directory for artifact generation

* Flat event based timer variation

Event based timer allows for easily starting and stopping timers without
wiping sub-timer data. It also requires less branching logic when
tracking time.

The json output is non-hierarchical in this version and hierarchy is
less rigidly enforced between starting and stopping.

* Add and write timers for top level install

* Update completion

* remove unused subtimer api

* Fix unit tests

* Suppress timing summary option

* Save timers summaries to user_data artifacts

* Remove completion from fish

* Move spack python to script section

* Write timer correctly for non-cache installs

* Re-add hash to timer file

* Fish completion updates

* Fix null timer yield value

* fix type hints

* Remove timer-summary-file option

* Add "." in front of non-package timer name

---------

Co-authored-by: Harmen Stoppels <harmenstoppels@gmail.com>
Co-authored-by: Harmen Stoppels <me@harmenstoppels.nl>

tldahlgren committed:
Docs/Packaging guide: Add BundlePackage (#39691)

* Docs/Packaging guide: Add BundlePackage
* Adjusted version ordering note to reflect convention.

dependabot[bot] committed:
build(deps): bump pytest from 7.4.1 to 7.4.2 in /lib/spack/docs (#39883)

Bumps [pytest](https://github.com/pytest-dev/pytest) from 7.4.1 to 7.4.2.
- [Release notes](https://github.com/pytest-dev/pytest/releases)
- [Changelog](https://github.com/pytest-dev/pytest/blob/main/CHANGELOG.rst)
- [Commits](https://github.com/pytest-dev/pytest/compare/7.4.1...7.4.2)

---
updated-dependencies:
- dependency-name: pytest
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

alalazo committed:
crosstool-ng: add new package (#39865)

alalazo committed:
Speed-up `spack external find` execution (#39843)

* Perform external spec detection with multiple workers

The logic to perform external spec detection has been refactored
into classes. These classes use the GoF "template" pattern to account
for the small differences between searching for "executables" and
for "libraries", while unifying the larger part of the algorithm.

A ProcessPoolExecutor is used to parallelize the work.

* Speed-up external find by tagging detectable packages automatically

Querying packages by tag is much faster than inspecting the repository,
since tags are cached. This commit adds a "detectable" tag to every
package that implements the detection protocol, and external detection
uses it to search for packages.

* Pass package names instead of package classes to workers

The slowest part of the search is importing the Python modules
associated with candidate packages. The import is done serially
before we distribute the work to the pool of executors.

This commit pushes the import of the Python module to the job
performed by the workers, and passes just the name of the packages
to the executors.

In this way imports can be done in parallel.

* Rework unit-tests for Windows

Some unit tests were doing a full e2e run of a command
just to check a input handling. Make the test more
focused by just stressing a specific function.

Mark as xfailed 2 tests on Windows, that will be fixed
by a PR in the queue. The tests are failing because we
monkeypatch internals in the parent process, but the
monkeypatching is not done in the "spawned" child
process.

toxa81 committed:
BUILD_TESTING is a CMake variable and should not be scoped (#39866)

tgamblin committed:
bugfix: only complete aliases for potential aliases (#39887)

Smart alias completion introduced in #39499 wasn't as smart as it needed to be, and
would complete any invalid command prefix and some env names with alias names.

- [x] don't complete aliases if there are no potential completions
      e.g., don't convert `spack isnotacommand` -> `spack concretize`

- [x] don't complete with an aliases if we're not looking at a top-level subcommand.

danlipsa committed:
Windows decompression: fix removal of intermediate file (#38958)

Extensionless archives requiring two-stage decompression and extraction
require intermediate archives to be renamed after decompression/extraction
to prevent collision. Prior behavior attempted to cleanup the intermediate
archive with the original name, this PR ensures the renamed folder is
cleaned instead.

Co-authored-by: Dan Lipsa <dan.lipsa@khq.kitware.com>
Co-authored-by: John Parent <john.parent@kitware.com>

amd-toolchain-support committed:
Cloverleaf-ref: proposal for a -ref version only (#39894)

* Cloverleaf-ref: proposal for a -ref version only
* Cloverleaf-ref: fixing typo.

---------

Co-authored-by: fpanichi <fpanichi@amd.com>

johnwparent committed:
Spack on Windows: fix shell scripts when root contains a space (#39875)

Enclose variable tracking root in quotes.

eugeneswalker committed:
veloc: add v1.7 (#39465)

* updates for VELOC 1.6

* veloc: add v1.7

---------

Co-authored-by: Bogdan Nicolae <bogdan.nicolae@acm.org>

johnwparent committed:
NMake Builder: change property name (#39824)

NMake makefiles are still called makefiles. The corresponding builder
variable was called "nmakefile", which is a bit unintuitive and lead
to a few easy-to-make, hard-to-notice mistakes when creating packages.
This commit renames the builder property to be "makefile"

tmadlener committed:
root: add latest available tag (#39895)

dependabot[bot] committed:
build(deps): bump actions/upload-artifact from 3.1.2 to 3.1.3 (#39852)

Bumps [actions/upload-artifact](https://github.com/actions/upload-artifact) from 3.1.2 to 3.1.3.
- [Release notes](https://github.com/actions/upload-artifact/releases)
- [Commits](https://github.com/actions/upload-artifact/compare/0b7f8abb1508181956e8e162db84b466c27e18ce...a8a3f3ad30e3422c9c7b888a15615d19a852ae32)

---
updated-dependencies:
- dependency-name: actions/upload-artifact
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps): bump docker/build-push-action from 4.1.1 to 4.2.1 (#39901)

Bumps [docker/build-push-action](https://github.com/docker/build-push-action) from 4.1.1 to 4.2.1.
- [Release notes](https://github.com/docker/build-push-action/releases)
- [Commits](https://github.com/docker/build-push-action/compare/2eb1c1961a95fc15694676618e422e8ba1d63825...0a97817b6ade9f46837855d676c4cca3a2471fc9)

---
updated-dependencies:
- dependency-name: docker/build-push-action
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

RMeli committed:
spglib: add v2.1.0 (#39910)

* ensure umpire~cuda~rocm when ~cuda~rocm

* update spglib

markcmiller86 committed:
Update package H5Z-ZFP to 1.1.1 (#39740)

* update to 1.1.1

* add maintainers

* add 1.1.0; set default

ledif committed:
singularity: fix package name in comment (#39822)

freifrauvonbleifrei committed:
DisCoTec: add new package (#35239)


Co-authored-by: Alexander Van Craen <40516079+vancraar@users.noreply.github.com>
Co-authored-by: Massimiliano Culpo <massimiliano.culpo@gmail.com>

rscohn2 committed:
intel-oneapi-mkl: hpcx-mpi support (#39750)

annop-w committed:
wrf: fix patches for 4.2 to 4.4.2 for aarch64 config (#39900)

A-N-Other committed:
gffcompare: new package @0.12.6 (#39917)

Co-authored-by: LMS Bioinformatics <bioinformatics@lms.mrc.ac.uk>

A-N-Other committed:
transdecoder: add 5.7.1 (#39916)

Co-authored-by: LMS Bioinformatics <bioinformatics@lms.mrc.ac.uk>

kjrstory committed:
openfoam-org: zoltan renumbering and decompsition (#39915)

TomTheBear committed:
likwid: search library path for compilation with hwloc (#39659)

mirguest committed:
Garfield++: Add the patch to fix the missing headers and Set the environment variable HEED_DATABASE (#39904)

AlexanderRichert-NOAA committed:
sp: add develop version, build precision, and OpenMP support (#39874)

Co-authored-by: Tamara Dahlgren <35777542+tldahlgren@users.noreply.github.com>

annop-w committed:
acfl: map rocky9 to RHEL-9 (#39892)

AlexanderRichert-NOAA committed:
w3emc: add develop version and more variants (#39884)

Co-authored-by: Massimiliano Culpo <massimiliano.culpo@gmail.com>

scottwittenburg committed:
gitlab ci: Use image with awscli for rebuild-index job (#39636)

Instead of pointing to the image on DockerHub, which rate limits us and
causes pipeline failures durying busy times, use the version at ghcr.

And we might as well use the ghcr version everywhere else too.

simonpintarelli committed:
sirius: fix cuda_arch flags (#39889)

sethrj committed:
celeritas: add v0.3.2 (#39879)

payerle committed:
elpa: Add --enable-scalapack-tests to configure when +autotune (#39728)

When building elpa +autotune, we apparently need to add
the configure option --enable-scalapack-tests

mtaillefumier committed:
Support of versions of cp2k below 7 marked as deprecated (#39652)

SIRIUS was introduced in version 7 of cp2k but could be used
in practice in version 9 (input format and functionalities).
SIRIUS with version 6 and below are marked as a dependency
conflict until CP2K version 9.

Co-authored-by: Mikael Simberg <mikael.simberg@iki.fi>

alalazo committed:
Update archspec to latest commit (#39920)

- [x] Intel flags for old architectures
- [x] Support for Sapphire Rapids
- [x] Cache the "ancestors" computation

haampie committed:
glibc: fix deps, drop pre 2.17 (#39806)

* glibc: add missing deps, drop < 2.17

Building glibc 2.17 requires linking libgcc_{s,eh}.a, which themselves
depend on whatever glibc the current gcc uses for its runtime libraries.

Newer gcc depends on gnu extensions of libdl it seems, so that means you
simply cannot build old glibc with gcc-using-new-glibc.

The relevant fix in glibc is this commit:

commit 95f5a9a866695da4e038aa4e6ccbbfd5d9cf63b7
Author: Joseph Myers <joseph@codesourcery.com>
Date:   Tue Jul 3 19:14:59 2012 +0000

    Avoid use of libgcc_s and libgcc_eh when building glibc.

See also references to that commit in the glibc mailing list.

* update the gmake bound

* add --without-selinux

manuelakuhn committed:
py-dunamai: add 1.18.0 (#39849)

svalat committed:
NUMAPROF: add v1.1.5 (#39891)

svalat committed:
MALT: add v1.2.2 (#39890)

dslarm committed:
ctffind: extend a Power 9 patch to AArch64 (#39137)

tmadlener committed:
dd4hep: make sure to find libraries correctly (#39516)

tdrayna committed:
dakota: add v6.18 (#39882)


Co-authored-by: Travis Drayna <tdrayna@loki.local>

Barbozafab committed:
py-enum-tools: add new package and dependencies (#39672)

* Adds package 'enum-tools'

* Fixes syntax

* py-enum-tools: update pypi link and dependencies

* py-whey: add new package

* py-consolekit: add new package

* py-shippinglabel: add new package

* py-project-parser: add new package

* py-handy-archives: add new package

* py-domdf-python-tools: add new package

* py-dom-toml: add new package

* py-dist-meta: add new package

* py-deprecation-alias: add new package

* py-pygments: add versions 2.16.0 and 2.16.1

* py-mistletoe: add new package

* py-apeye: add new package

* py-apeye-core: add new package

* Fix dependencies' names and remove redundancies

* Fix style

* py-enum-tools: fix dependencies

* py-mistletoe: update dependencies

* Apply suggestions from code review

Co-authored-by: Adam J. Stewart <ajstewart426@gmail.com>

* py-whey: fix dependencies

* py-whey: fix style

* py-apeye: fix dependencies

---------

Co-authored-by: Fábio de Andrade Barboza <f168817@dac.unicamp.br>
Co-authored-by: Adam J. Stewart <ajstewart426@gmail.com>

joseeroman committed:
SLEPc: add v3.19.2 (#39805)

Co-authored-by: Satish Balay <balay@mcs.anl.gov>

Barbozafab committed:
py-pipdeptree: add new package; py-hatchling: add 1.18.0 (#39697)

* py-hatchling: add 1.18.0

* py-pipdeptree: add new package

* py-hatchling: add Python 3.8 dependency

* Apply suggestion from code review

Co-authored-by: Adam J. Stewart <ajstewart426@gmail.com>

---------

Co-authored-by: Fábio de Andrade Barboza <f168817@dac.unicamp.br>
Co-authored-by: Adam J. Stewart <ajstewart426@gmail.com>

scothalverson committed:
gdrcopy: inherit from CudaPackage, update dependencies (#39596)

jmcarcell committed:
Add another url for whizard (#37354)

Co-authored-by: jmcarcell <jmcarcell@users.noreply.github.com>

simonpintarelli committed:
update q-e-sirius package (#39744)

RMeli committed:
Update MDAnalysis (#39787)

* ensure umpire~cuda~rocm when ~cuda~rocm

* update mdanalysis

* Update var/spack/repos/builtin/packages/py-mdanalysis/package.py

Co-authored-by: Adam J. Stewart <ajstewart426@gmail.com>

* Update var/spack/repos/builtin/packages/py-mdanalysis/package.py

---------

Co-authored-by: Adam J. Stewart <ajstewart426@gmail.com>

nilsvu committed:
spectre: add v2023.08.18 and v2023.09.07 (#39925)

sidpbury committed:
py-fairscale and dependencies py-pgzip libgit2 py-pygit2 (#39220)

* simple build of py-openai

* added variants to py-openai

* py-pandas-stubs is a dependency for py-openai

* fixed format and flake8 errors for py-openai

* black format error for py-pandas-stubs

* [@spackbot] updating style on behalf of sidpbury

* made style and format changes to py-openai

* made style and format changes to py-pandas-stubs

* py-types-pytz is a dependency for py-openai

* [@spackbot] updating style on behalf of sidpbury

* updated py-openpyxl for ver 3.0.7 and 3.1.2

* Update var/spack/repos/builtin/packages/py-pandas-stubs/package.py

Co-authored-by: Adam J. Stewart <ajstewart426@gmail.com>

* ajs requested changes for py-openai

* updated py-openpyxl for supported python

* [@spackbot] updating style on behalf of sidpbury

* updated py-openpyxl

* removed requirement.txt dependencies in  py-openpyxl

* removed python depends on from openpyxl

* updated package to support newer versions

* updated version of py-pygit2

* py-fairscale is a new package in support of torch

* py-pgzip is a dependency for py-fairscale

* switch fairscale pypi, added extra variant for convenience

* removed python dependency

* changed multiple requirement versions

* changes for upstream py-fairscale

* changes for upsteam py-pygit2

* Update package.py

* Update package.py

* sorted out some of the dependency versions

* removed version 1.12.2 because dependency could not be met

* updated py-cached-property dependency

* suggested changes from adamjstewart

---------

Co-authored-by: Adam J. Stewart <ajstewart426@gmail.com>

rbberger committed:
cuda: add versions 12.2.0 and 12.2.1 (#39684)

hainest committed:
fpchecker: add version 0.4.0 (#39924)

mikerenfro committed:
correct py-bx-python@0.8.8 python dependency (#39766)

* correct python dependency for 0.8.8

* add missing zlib dependency

* restore missing whitespace

* correct versions for 0.9.0, clarify rules for future releases

* Update package.py

* Remove whitespace for black check

RMeli committed:
tidynamics: add v1.1.2 (#39911)

* ensure umpire~cuda~rocm when ~cuda~rocm

* update tydynamics

manuelakuhn committed:
py-tomlkit: add 0.12.1 (#39922)

manuelakuhn committed:
py-tinycss2: add 1.2.1 (#39921)

manuelakuhn committed:
py-tifffile: add 2023.8.30 (#39918)

adamjstewart committed:
py-black: add v23.9 (#39931)

Sinan81 committed:
package_qscintilla_build_with_qt6 (#39544)

* package_qscintilla_build_with_qt6

* Update var/spack/repos/builtin/packages/qscintilla/package.py

Co-authored-by: Adam J. Stewart <ajstewart426@gmail.com>

* improve

* fix Qsci.abi3.so install

* simplify, fix, tidy

* fix style

* fix style

* fix style

* Update var/spack/repos/builtin/packages/qscintilla/package.py

Co-authored-by: Adam J. Stewart <ajstewart426@gmail.com>

* Update var/spack/repos/builtin/packages/qscintilla/package.py

Co-authored-by: Adam J. Stewart <ajstewart426@gmail.com>

* Update var/spack/repos/builtin/packages/qscintilla/package.py

Co-authored-by: Adam J. Stewart <ajstewart426@gmail.com>

* Update var/spack/repos/builtin/packages/qscintilla/package.py

Co-authored-by: Adam J. Stewart <ajstewart426@gmail.com>

* Update var/spack/repos/builtin/packages/qscintilla/package.py

Co-authored-by: Adam J. Stewart <ajstewart426@gmail.com>

* improve

* improve style

* fix style

* make black happy

* add ver 2.14.1

* update

* make black happy

* Update var/spack/repos/builtin/packages/qscintilla/package.py

Co-authored-by: Adam J. Stewart <ajstewart426@gmail.com>

* improve

---------

Co-authored-by: Sinan81 <Sinan@world>
Co-authored-by: Adam J. Stewart <ajstewart426@gmail.com>

climbfuji committed:
Add awscli@1.29.41 and its dependency py-botocore@1.31.41 (#39878)

* Add awscli@1.29.41 and its dependency py-botocore@1.31.41

* Update var/spack/repos/builtin/packages/awscli/package.py

Co-authored-by: Manuela Kuhn <36827019+manuelakuhn@users.noreply.github.com>

---------

Co-authored-by: Manuela Kuhn <36827019+manuelakuhn@users.noreply.github.com>

adamjstewart committed:
py-build: add v1 (#39930)

cmelone committed:
fish: add 3.6.0, 3.6.1, update cmake requirement (#39899)

stephenmsachs committed:
[gromacs] Fix 2021.2 build error (#39722)

* [gromacs] Fix 2021.2 build error

See https://gromacs.bioexcel.eu/t/compilation-failure-for-gromacs-2021-1-and-2021-2-with-cmake-3-20-2/2129

* Update var/spack/repos/builtin/packages/gromacs/package.py

Co-authored-by: Andrey Alekseenko <al42and@gmail.com>

---------

Co-authored-by: Andrey Alekseenko <al42and@gmail.com>

bvanessen committed:
lbann: make two variants sticky (#39710)

* Make the state of the Python Front End (PFE) and Python data reader
support sticky so that the concretizer does not arbitrarily disable
them.

Co-authored-by: Massimiliano Culpo <massimiliano.culpo@gmail.com>

dependabot[bot] committed:
build(deps): bump black from 23.7.0 to 23.9.1 in /lib/spack/docs (#39937)

Bumps [black](https://github.com/psf/black) from 23.7.0 to 23.9.1.
- [Release notes](https://github.com/psf/black/releases)
- [Changelog](https://github.com/psf/black/blob/main/CHANGES.md)
- [Commits](https://github.com/psf/black/compare/23.7.0...23.9.1)

---
updated-dependencies:
- dependency-name: black
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

A-N-Other committed:
filtlong: add v0.2.1, patch for %gcc@13: (#39775)

Co-authored-by: LMS Bioinformatics <bioinformatics@lms.mrc.ac.uk>

msimberg committed:
Add gperftools 2.13 (#39940)

haampie committed:
asp.py: fix deprecation warning (#39943)

haampie committed:
openssl: deprecate 1.1.1 (#39942)

vincentmr committed:
Update PennyLane packages to v0.32. (#39688)

* Update PennyLane packages to v0.32.

* Reformat.

* Couple small fixes.

* Fix Lightning cmake_args.

* Couple dep fixes in py-pennylane + plugins.

* Fix scipy condition.

* Add comment on conflicting requirement.

aumuell committed:
openexr: add 2.4.3, 2.5.9, 3.1.11 & 3.2.0 (#39863)

* openexr: add 2.4.3, 2.5.9, 3.1.11 & 3.2.0

- 2.5.9 is the latest version compatible with OpenScenGraph
- improved compatibility with GCC 13

Co-authored-by: Adam J. Stewart <ajstewart426@gmail.com>

* openexr: drop unsupported debug variant

--disable/enable-debug is not supported by ./configure in 1.3, 1.7, 2.0, 2.2 and 2.3

* openexr: transform into multi-build system package

simplifies package considerably, as nothing special seems to be required

* openexr: pkg-config is also used by @3

* openexr: use system libdeflate instead of internal

if no libdeflate is found, openexr would download and build its own starting with 3.2.0

* openexr: disable tests

would download lots of data during cmake and make build times noticably longer

---------

Co-authored-by: Adam J. Stewart <ajstewart426@gmail.com>

WeiqunZhang committed:
amrex: add v23.09 (#39788)

AlexanderRichert-NOAA committed:
g2 package bugfix (#39923)

* g2 package bugfix

* g2: fix w3emc dep

* fix setup_run_environment

* Update package.py

* Update package.py

* Update package.py

* Update package.py

* Update package.py

* style fix

* Update var/spack/repos/builtin/packages/g2/package.py

Co-authored-by: Massimiliano Culpo <massimiliano.culpo@gmail.com>

* Update var/spack/repos/builtin/packages/g2/package.py

Co-authored-by: Massimiliano Culpo <massimiliano.culpo@gmail.com>

---------

Co-authored-by: Massimiliano Culpo <massimiliano.culpo@gmail.com>

climbfuji committed:
Filter spack compiler wrappers for h4cc in var/spack/repos/builtin/packages/hdf/package.py (#39870)

climbfuji committed:
libjpeg-turbo: Fix Darwin lib install name (#39834)

* Fix Darwin lib install name in var/spack/repos/builtin/packages/libjpeg-turbo/package.py

* Update var/spack/repos/builtin/packages/libjpeg-turbo/package.py

Co-authored-by: Tamara Dahlgren <35777542+tldahlgren@users.noreply.github.com>

---------

Co-authored-by: Tamara Dahlgren <35777542+tldahlgren@users.noreply.github.com>

adammoody committed:
mpifileutils: switch to main branch (#39525)

tldahlgren committed:
CI/Add superlu-dist to broken stand-alone tests due to hang (#38908)

* Add superlu-dist to broken stand-alone CI tests
* Revert "disable superlu test (#38894)"
  This reverts commit e16397b5d865d402c749482d05bb2bdc09b96661.

A-N-Other committed:
bam-readcount: add 1.0.1, correct build flags (#39950)

Co-authored-by: LMS Bioinformatics <bioinformatics@lms.mrc.ac.uk>

alalazo committed:
binutils: add v2.41 (#39483)

jgaeb committed:
r-pl94171: Add R package PL94171 (#39926)

jgaeb committed:
r-piggyback: Add `R` package `piggyback` (#39932)

scottwittenburg committed:
ci: Tag reindex job to match the image architecture (#39958)

alalazo committed:
Fix typo in metaclasses (#39947)

adamjstewart committed:
py-lightly: add v1.4.18 (#39960)

AlexanderRichert-NOAA committed:
Update g2c recipe: add develop version and misc. variants (#39965)

* Update g2c recipe: add develop version and misc. variants
* add testing to g2c
* Update package.py

haampie committed:
py-maestrowf: new version, fix artificial py-cryptography dep, support openssl 3 (#39946)

* py-maestrowf: add new version 1.1.9, deprecate development releases

* py-maestrowf: drop py-cryptography in 1.1.9

* py-maestrowf: drop py-cryptography dependency entirely, since it is not a direct dependency

* py-merlin: new version, ensure openssl 3 compat

* py-merlin: drop py-coloredlogs@10: lowerbound

* py-maestrowf: add py-rich, reorder deps

* py-celery: explain why upperbound is in spack but not in requirements.txt

haampie committed:
unifyfs: support openssl 3 (#39945)

* unifyfs: drop upperbound on deprecated openssl

The package uses deprecated MD5 functions from OpenSSL, which causes
warnings, but (a) Spack by default disables -Werror, and (b) those
functions will continue to exist in OpenSSL 3.

* unifyfs: enable parallel build, only make check sequential

* unifyfs: order class methods by install phases

haampie committed:
gitlab ci: temporarily disable darwin aarch64 (#39974)

as there are no or few runners available

alalazo committed:
Spec.tree: add type hints (#39948)

marcmengel committed:
Merge branch 'per_os_config_feature' of github.com:marcmengel/spack into per_os_config_feature

marcmengel commented:
Rebased off of develop...
